runtime/autoload/provider/pythonx.vim

```diff
@@ -39,31 +44,33 @@ function! s:check_version(prog, ver, skip) abort
     return [1, '']
   endif
 
-  let get_version =
-        \ ' -c "import sys; sys.stdout.write(str(sys.version_info[0]) + '.
-        \ '\".\" + str(sys.version_info[1]))"'
-  let min_version = (a:ver == 2) ? '2.6' : '3.3'
-  if system(a:prog . get_version) >= min_version
+  let min_version = (a:ver[0] == 2) ? '2.6' : '3.3'
+  if a:ver >= min_version
     return [1, '']
   endif
-  return [0, 'Python ' . get_version . ' interpreter is not supported.']
+  return [0, 'Python ' . a:ver . ' interpreter is not supported.']
 endfunction
 
 function! s:check_interpreter(prog, ver, skip) abort
   if !executable(a:prog)
-    return [0, 'Python'.a:ver.' interpreter is not executable.']
+    return [0, 'Python'.a:ver.' interpreter is not executable.', '']
   endif
 
   if a:skip
-    return [1, '']
+    return [1, '', '']
   endif
 
-  " Load neovim module check
-  call system(a:prog . ' -c ' .
+  " Try to load neovim module, and output Python version.
+  let ver = system(a:prog . ' -c ' .
+        \ '''import sys; sys.stdout.write(str(sys.version_info[0]) + '.
+        \ '"." + str(sys.version_info[1])); '''.
         \ (a:ver == 2 ?
         \   '''import pkgutil; exit(pkgutil.get_loader("neovim") is None)''':
         \   '''import importlib; exit(importlib.find_loader("neovim") is None)''')
         \ )
```

Comment:
Not necessarily for this PR, but since this python snippet is growing wouldn't it be better to save it as python_host_check.py in runtime and just `system(a:prog . ' $VIMRUNTIME/.../python_host_check.py')`? It could even print out the python-client version too.